runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -195,7 +195,7 @@ function TSHighlighter:destroy()
       api.nvim_buf_call(self.bufnr, function()
         api.nvim_exec_autocmds(
           'FileType',
-          { group = 'syntaxset', buffer = self.bufnr, modeline = false }
+          { group = 'syntaxset', buf = self.bufnr, modeline = false }
```

Comment:
API functions like `nvim_buf_set_keymap` and `nvim_exec_autocmds` will need to support both `buf` and `buffer` (for backwards compatibility).

But all the Lua things that wrap them can just pass `buf` to the API functions; the Lua wrappers don't need to explicitly support both.